File: Javascript/CRUDJavascriptShowTotalInTerminal/Controller/ProductController.js
const ProductCrud = require("../Model/ProductCrud");

class ProductController {

    constructor() {
        this.crud = new ProductCrud();
    }

    add(product) {
        this.crud.addProduct(product);
    }

    getAll() {
        return this.crud.getAllProducts();
    }

    update(id, product) {
        return this.crud.updateProduct(id, product);
    }

    delete(id) {
        return this.crud.deleteProduct(id);
    }

    calculateTotal() {
        const products = this.crud.getAllProducts();
        let total = 0;

        for (let i = 0; i < products.length; i++) {
            total += products[i].price * products[i].quantity;
        }

        return total;
    }
}

module.exports = ProductController;

File: EjemplosJson/Python/CRUDPythonProducts/Product_controller.py
from ProductCrud_model import ProductCrud

class ProductController:
    def __init__(self):
        self.crud = ProductCrud()

    def add(self, product):
        self.crud.add_product(product)

    def get_all(self):
        return self.crud.get_all_products()

    def update(self, id, product):
        return self.crud.update_product(id, product)

    def delete(self, id):
        return self.crud.delete_product(id)

    def calculate_total(self):
        products = self.crud.get_all_products()
        total = 0

        for product in products:
            total += product["price"] * product["quantity"]

        return total
